src/SearchService.php

```diff
@@ -99,4 +105,23 @@ protected function formatFileSizeString(string $fileSizeBytes): string
 
         return round($fileSizeBytes, 0) . $units[$prefixIndex];
     }
+
+    /**
+     * The functions takes in a full url (including protocol) and tries to find a
+     * matching file in the assets folder
+     *
+     *
+     * @param string $url
+     * @return File|null
+     */
+    private function getFileFromURL(string $url): ?File
+    {
+        $parts =   array_filter(preg_split("#[/\\\\]+#", $url ?? '') ?? []);
```

Comment:
Instead of doing this, I'd suggest you use `parse_url()` instead maybe, then you can just extract the `PATH` part of the URL and let PHP handle the rest. You'd still have to strip off 'assets' I think separately.

Reply:
I thought about that, but we need to also get rid '/assets/'. File::find can't include the 'assets' part in the file path.

Reply:
Yeah, but this `preg_split` won't work correctly if the URL is slightly malformed. For example http://example.com//assets/file.pdf - in that case it won't unset the `assets` part of the URL.

Can I suggest we do `parse_url` first, then manually remove `ASSETS_DIR` using `substr` or similar? Thanks!